### PR TITLE
Design rehaul

### PR DESCRIPTION
**Pull request by kemcake**
Cherry-picking all design/ui work

### CODE REVIEW

**Comment by sebgoa** [comment]
let's try to get @Angelmmiguel to review.

File: src/ui/src/app/chart-details/chart-details.component.html [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
   <app-loader [loading]="loading">
     <section class="chart-details" *ngIf="chart">
-      <article class="chart-details__docs">
-        <app-chart-details-readme [chart]=chart [currentVersion]=currentVersion></app-chart-details-readme>
-      </article>
-      <aside class="chart-details__info">
-        <app-chart-details-info [chart]=chart [currentVersion]=currentVersion></app-chart-details-info>
-      </aside>
+      <div class="chart-details__header">
+        <div class="chart-details__header__background" [style.background-color]="chartColor"></div>
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.html]: I like this concept but we don't control the color of icons. If the "main" color of an icon is to bright or dark, the readability of the content will be affected. Also, if the color is too saturated, the general design will look weird for that page.

File: src/ui/src/app/chart-details/chart-details.component.scss [hunk 2]
```diff
@@ -3,31 +3,95 @@
...
-    &__info {
-      padding-left: 1em;
+  &__header {
+    width: 100%;
+    margin-top: -80px;
+    margin-bottom: 40px;
+    &__background {
+      width: 100%;
+      height: 200px;
+      background: md-color($monocular-app-primary);
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.scss]: This color will be overridden by the Icon's color when it's calculated. This will cause a flashing effect in the UI.

File: src/ui/src/app/chart-details/chart-details.component.scss [hunk 3]
```diff
@@ -3,31 +3,95 @@
...
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      img {
+        width: 70%;
+        max-height: 70%;
+      }
     }
+  }
+  &__content {
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.scss]: This selector is duplicated on this file.

File: src/ui/src/app/chart-details/chart-details.component.scss [hunk 4]
```diff
@@ -3,31 +3,95 @@
...
+    max-width: 1000px;
+    margin: auto;
+    display: flex;
+    flex-direction: column-reverse;
+    padding: 0 1em;
+    padding-bottom: 1em;
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.scss]: Merge this value in the previous CSS property

File: src/ui/src/app/chart-details/chart-details.component.scss [hunk 5]
```diff
@@ -3,31 +3,95 @@
...
-    flex-direction: row;
-    flex-wrap: nowrap;
-
-    &__info {
-      padding-left: 1em;
+  &__header {
+    width: 100%;
+    margin-top: -80px;
+    margin-bottom: 40px;
+    &__background {
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.scss]: For readability of the CSS code, can you add an space before every selector? It's helpful to follow the hierarchy in the files :)

File: src/ui/src/app/chart-details/chart-details.component.scss [hunk 6]
```diff
@@ -3,31 +3,95 @@
...
+      width: 100%;
+      height: 200px;
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.scss]: Can you move fixed heights and widths to variables at the top of the file?

File: src/ui/src/app/chart-details/chart-details.component.scss [hunk 7]
```diff
@@ -3,31 +3,95 @@
...
+      margin-top: -75px;
+      display: flex;
     }
-
-    &__info {
-      width: 35%;
+    &__text {
+      flex: 1;
+      h1 {
+        height: 70px;
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.scss]: Why you force the height of title element? If the user increase the zoom of the page this section won't grow as well

File: src/ui/src/app/chart-details/chart-details.component.scss [hunk 8]
```diff
@@ -3,31 +3,95 @@
...
+        color: white;
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.scss]: Try to use colors from the `theme.scss` file

File: src/ui/src/app/chart-item/chart-item.component.scss [hunk 9]
```diff
@@ -4,41 +4,63 @@
...
+    transform: scale(0.98);
+  }
   &-content {
-    text-align: center;
+    height: $chart-item-content-height;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    background: md-color($monocular-app-primary);
+    color: white;
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-item/chart-item.component.scss]: Try to use colors from the `theme.scss` file

File: src/ui/src/app/chart-item/chart-item.component.scss [hunk 10]
```diff
@@ -4,41 +4,63 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-end;
+    word-wrap: break-word;
+    overflow: hidden;
     &__title {
-      color: md-color($monocular-app-primary);
-      margin: 0 0 .5em;
-      font-weight: 300;
+      a { color: white; }
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-item/chart-item.component.scss]: Don't contract selectors. This will be optimized automatically

File: src/ui/src/app/chart-item/chart-item.component.scss [hunk 11]
```diff
@@ -63,10 +86,6 @@ $chart-item-content-height: 80px;
     }
 
     &__version {
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-item/chart-item.component.scss]: Remove selectors without any CSS property

File: src/ui/src/app/chart-details/chart-details.component.ts [hunk 12]
```diff
@@ -59,4 +67,28 @@ export class ChartDetailsComponent implements OnInit {
...
+
+  getIconUrl(): string {
+    let icons = this.chart.relationships.latestChartVersion.data.icons;
+    if (icons !== undefined && icons.length > 0) {
+      const icon =
+        this.config.backendHostname +
+        icons.find(icon => icon.name === '160x160-fit').path;
+      if (!this.chartColor) {
+        RGBaster.colors(icon, {
+          success: payload => {
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.ts]: You can compact this line:

```
success: payload => this.chartColor = payload.best
```

File: src/ui/src/app/chart-item/chart-item.component.ts [hunk 13]
```diff
@@ -40,7 +43,18 @@ export class ChartItemComponent implements OnInit {
   getIconUrl(): string {
     let icons = this.chart.relationships.latestChartVersion.data.icons;
     if (icons !== undefined && icons.length > 0) {
-      return this.config.backendHostname + icons.find(icon => icon.name === '160x160-fit').path;
+      const icon =
+        this.config.backendHostname +
+        icons.find(icon => icon.name === '160x160-fit').path;
+      if (!this.backgroundColor) {
+        RGBaster.colors(icon, {
+          success: payload => {
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-item/chart-item.component.ts]: You can compact this line:

```
success: payload => this.chartColor = payload.best
```

File: src/ui/src/app/chart-search/chart-search.component.ts [hunk 14]
```diff
@@ -1,56 +0,0 @@
-import { Component, OnInit } from '@angular/core';
```
Comment (Angelmmiguel) [inline src/ui/src/app/chart-search/chart-search.component.ts]: Why did you remove these files? The purpose of `search` components is to create a specific view with an independent URL. Users can link to an specific search.

File: src/ui/src/app/charts-filters/charts-filters.component.ts [hunk 15]
```diff
@@ -1,48 +0,0 @@
-import { Component, Input, OnInit, Output, EventEmitter } from '@angular/core';
```
Comment (Angelmmiguel) [inline src/ui/src/app/charts-filters/charts-filters.component.ts]: I see you added all the filter logic to `charts` component. Now, the `charts` component is bigger and I don't see any advantages to merge both files instead of having two separate components.

File: src/ui/src/app/charts/charts.component.scss [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  &__gallery {
+    width: 100%;
+    display: flex;
+    flex-direction: row;
+
+    &__sidebar {
+      width: 175px;
+      border-right: 1px solid #D7D9DD;
```
Comment (Angelmmiguel) [inline src/ui/src/app/charts/charts.component.scss]: Use colors from the `theme.scss` file.

File: src/ui/src/app/charts/charts.component.scss [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+      flex-direction: column;
+      app-chart-list {
+        flex: 1;
+        display: flex;
+        padding-left: 1em;
+      }
+      &__topbar {
+        display: flex;
+        height: 40px;
+        border-bottom: 1px solid #D7D9DD;
```
Comment (Angelmmiguel) [inline src/ui/src/app/charts/charts.component.scss]: Use colors from the `theme.scss` file

File: src/ui/src/app/charts/charts.component.scss [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+        align-items: center;
+        padding-left: 10px;
+        * { fill: rgba(0, 0, 0, 0.5); }
```
Comment (Angelmmiguel) [inline src/ui/src/app/charts/charts.component.scss]: Don't contract selectors

File: src/ui/src/app/charts/charts.component.scss [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+        input {
+          width: 100%;
+          height: 100%;
+          border: 0;
+          font-size: 18px;
```
Comment (Angelmmiguel) [inline src/ui/src/app/charts/charts.component.scss]: Use `em` units for font sizes.

File: src/ui/src/app/deployment-item/deployment-item.component.scss [hunk 20]
```diff
@@ -0,0 +1,60 @@
+// Import your custom theme
+@import '../../theme.scss';
+
+.deployment_item {
+  width: 100%;
+  height: 100%;
+  display: flex;
+  align-items: center;
+  background: #EAEDEF;
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployment-item/deployment-item.component.scss]: Use colors from the `theme.scss` file

File: src/ui/src/app/deployment-item/deployment-item.component.scss [hunk 21]
```diff
@@ -0,0 +1,60 @@
+// Import your custom theme
+@import '../../theme.scss';
+
+.deployment_item {
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployment-item/deployment-item.component.scss]: Some of the selectors of this class are pretty similar to the `chart-details`. Can we extract common parts to new components? If we create them, we won't need to duplicate the logic and UI.

File: src/ui/src/app/deployment-item/deployment-item.component.scss [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+      b {
+        background: rgba(0, 0, 0, 0.1);
+        border-radius: 4px;
+        padding: 1px 4px;
+        font-size: 0.9em;
+        font-weight: 600;
+      }
+    }
+    &__date {
+      position: absolute;
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployment-item/deployment-item.component.scss]: Is required to use a `absolute` positioning here? I don't see any media query in this file and Monocular was optimized to work in mobile too. `absolute` positioning may break the design in mobile screens

File: src/ui/src/app/deployment-item/deployment-item.component.html [hunk 23]
```diff
@@ -0,0 +1,20 @@
+<a class="deployment_item" href="/deployments/{{ deployment.id }}" routerLink="/deployments/{{ deployment.id }}"
+  [style.background-color]="backgroundColor"
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployment-item/deployment-item.component.html]: Same here about the issue of set a backgroundColor based on the icon color. We don't control it.

File: src/ui/src/app/deployment/deployment.component.scss [hunk 24]
```diff
@@ -1,38 +1,32 @@
 // Import your custom theme
 @import '../../theme.scss';
 
+.deployment-details__header {
+  width: 100%;
+  max-width: 1000px;
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployment/deployment.component.scss]: I see this `max-width: 1000px` property in a few CSS files. Can we move it to a common place?

File: src/ui/src/app/deployment/deployment.component.scss [hunk 25]
```diff
@@ -48,46 +42,70 @@
...
   }
 
   &__info {
-    &__title {
-      color: md-color($monocular-app-primary);
-      margin: 0 0 .5em;
-      font-weight: 300;
+
+    &__container {
+      background: #EAEDEF;
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployment/deployment.component.scss]: Use colors from the `theme.scss` file

File: src/ui/src/app/deployments/deployments.component.scss [hunk 26]
```diff
@@ -1,44 +1,93 @@
...
-    md-card-title {
-      color: md-color($monocular-app-primary);
-    }
-
-    md-icon {
-      color: darken($layout_light, 20%);
+    display: flex;
+    &__sidebar {
+      width: 175px;
+      border-right: 1px solid #D7D9DD;
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployments/deployments.component.scss]: Use colors from the `theme.scss` file

File: src/ui/src/app/deployments/deployments.component.scss [hunk 27]
```diff
@@ -1,44 +1,93 @@
...
-      md-card {
-        background-color: $layout_light;
+    &__content {
+      flex: 1;
+      display: flex;
+      flex-direction: column;
+      &__topbar {
+        display: flex;
+        height: 40px;
+        border-bottom: 1px solid #D7D9DD;
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployments/deployments.component.scss]: Use colors from the `theme.scss` file

File: src/ui/src/app/deployments/deployments.component.ts [hunk 28]
```diff
@@ -22,26 +27,80 @@ export class DeploymentsComponent implements OnInit {
...
+    var list: string[] = ['All'];
+    this.deployments.forEach(dp => {
+      if (list.indexOf(dp.attributes.namespace) == -1) {
+        list.push(dp.attributes.namespace);
+      }
+    });
+    return list;
+  }
+
+  filterDeployments() {
```
Comment (Angelmmiguel) [inline src/ui/src/app/deployments/deployments.component.ts]: I think we can create a common filter component that generate the `select` elements and trigger an event when some of them change.

File: src/ui/src/app/header-bar/header-bar.component.scss [hunk 29]
```diff
@@ -29,51 +37,112 @@ $header-bar-padding-small: 1.5em;
...
     margin-top: .5em;
-
-    .md-input-wrapper {
-      margin: .4em 0;
+    background: rgba(255, 255, 255, 0.125);
+    border-radius: 4px;
+    transition: background 0.2s ease;
+    padding: 0.3em 0.5em;
+    svg {
+      fill: rgba(255, 255, 255, 0.6);
```
Comment (Angelmmiguel) [inline src/ui/src/app/header-bar/header-bar.component.scss]: Is this necessary? We can update the SVG file to use the color defined in `color` property instead of force the color in the fill property.

File: src/ui/src/app/header-bar/header-bar.component.scss [hunk 30]
```diff
@@ -29,51 +37,112 @@ $header-bar-padding-small: 1.5em;
 
   &__search {
     margin-top: .5em;
-
-    .md-input-wrapper {
-      margin: .4em 0;
+    background: rgba(255, 255, 255, 0.125);
```
Comment (Angelmmiguel) [inline src/ui/src/app/header-bar/header-bar.component.scss]: I'd rather have a base color and change the alpha using SASS methods

File: src/ui/src/assets/icons/search.svg [hunk 31]
```diff
@@ -0,0 +1,8 @@
+<svg height="32px" version="1.1" viewBox="0 0 32 32" width="32px" xmlns="http://www.w3.org/2000/svg" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (Angelmmiguel) [inline src/ui/src/assets/icons/search.svg]: Use `svgo` to reduce the size of the SVG file

File: src/ui/src/index.html [hunk 32]
```diff
@@ -46,25 +46,39 @@ <h1 id="app-name"></h1>
     </app-root>
   </div>
   <script>
-    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+  (function(i, s, o, g, r, a, m) {
```
Comment (Angelmmiguel) [inline src/ui/src/index.html]: Why did you uncompress the Google Analytics method and change the indentation of the script?

**Comment by Angelmmiguel** [comment]
Related to my comments about taking the color of the icons, this is the problem I see:

Locust card

Mongo details page

Also, what's the motivation to fix the size of the UI to `1000px`? I think it's too narrow for some of the pages like details. At the beginning of the project we decided to use the 100% of the screen.

Details view

Related to the cards, the icon is very relevant to identify a product. Now, the main element in the card is the color of the icon and they are smaller. Also, sometimes that color doesn't match with the real brand color.

Locus card

About the Home page, the cards have more margin on the right of the page:

Home page

**Comment by kemcake** [comment]
Thank you @Angelmmiguel , this is probably the best review I've ever seen. I definitely moved without being aware of some guidelines you use, I'll fix this first and then I reply to each question.

File: src/ui/src/app/chart-details/chart-details.component.scss (hunk 8 above)
Comment (kemcake) [inline src/ui/src/app/chart-details/chart-details.component.scss]: Ok I agree to uses variable when necessary, but a white is just a white, I could create a `$text-white` variable in theme.scss, but not sure if this is what we want...

File: src/ui/src/app/chart-search/chart-search.component.ts (hunk 14 above)
Comment (kemcake) [inline src/ui/src/app/chart-search/chart-search.component.ts]: You can still link directly to a search with `/charts?q=test`
And having an endpoint where you only have the result and do nothing else was kind of painful. We remove a route for simplicity, and the user is back on a page he already knows (charts-list) where he can filter/sort and keep searching

File: src/ui/src/app/deployment-item/deployment-item.component.scss (hunk 22 above)
Comment (kemcake) [inline src/ui/src/app/deployment-item/deployment-item.component.scss]: This is an absolute in a relative container, allowing the content of the item to be centered vertically while having the date always sticked in the corner. this is supposed to work just fine on mobile too

**Comment by Angelmmiguel** [comment]
Great @kemcake! Btw, the purpose of some requested changes is to be consistent with the rest of the project. If you have any suggestion about how to improve the current approach, please feel free to propose them so we can do a review. Thanks!

File: src/ui/src/app/chart-details/chart-details.component.scss (hunk 8 above)
Comment (Angelmmiguel) [inline src/ui/src/app/chart-details/chart-details.component.scss]: I'd rather have all the used colors in the same file. If we keep them in a single file it's easies to refactor the design in the future :)

File: src/ui/src/app/deployment-item/deployment-item.component.scss (hunk 22 above)
Comment (Angelmmiguel) [inline src/ui/src/app/deployment-item/deployment-item.component.scss]: :+1:

File: src/ui/src/app/chart-search/chart-search.component.ts (hunk 14 above)
Comment (Angelmmiguel) [inline src/ui/src/app/chart-search/chart-search.component.ts]: Okey! It makes sense. Thanks for the clarification

**Comment by kemcake** [comment]
So the remaining questions are:

- Should we use that color theming depending on the logo, looks nice most of the time but I agree it has some risks. The idea was to put more color and add some life to a chart. There's some checks on the color script to make sure we don't select too bright/dark color, but this is maybe not enough.

- max width layout is a common thing (github...), 100% looks really strange on a wide screen and is usually bad for readable stuff like the readme. Once again this is a suggestion, I moved the value to theme.scss so it's easily editable.

**Comment by sebgoa** [comment]
personally I am fine with the fixed width. I agree that on big screens it stretches things to a very weird point.

For the colors, I don't really care. The main problem is that we do not have logos for all charts.

@Angelmmiguel can you do a final review so that we can move forward.

This can always be revisited.

**Comment by Angelmmiguel** [comment]
@kemcake great job! Thanks for the improvements and refactoring. About the remaining items:

> Should we use that color theming depending on the logo, looks nice most of the time but I agree it has some risks. The idea was to put more color and add some life to a chart. There's some checks on the color script to make sure we don't select too bright/dark color, but this is maybe not enough.

I'd remove this feature. I prefer to have a controlled UI since the base color of the logos may not fit with the current design. Also, I think it's better to continue with the initial design of the cards. We did a research about the cards and decided to keep logos as the main and more important element.

> max width layout is a common thing (github...), 100% looks really strange on a wide screen and is usually bad for readable stuff like the readme. Once again this is a suggestion, I moved the value to theme.scss so it's easily editable.

That's solution works for me! Also, can you increase a bit that fixed width value? I believe that a value between 1150-1250px will work in most use cases.

Thanks!

**Comment by prydonius** [comment]
This looks great to me, only thing I noticed was the arrows in the Install section:
screen shot 2017-07-12 at 16 48 23

Oddly, if you click on "Helm CLI" it goes back to normal:

screen shot 2017-07-12 at 16 48 36

I don't mind if you prefer to merge this and fix this in a separate PR though.

**Comment by sebgoa** [comment]
@kemcake can you fix the conflicts, rebase and then we will merge.

**Comment by prydonius** [comment]
This worked before but I'm now unable to compile this again:

```
ui_1   | Time: 62932ms
ui_1   | chunk    {0} main.bundle.js, main.bundle.map (main) 591 kB {2} [initial] [rendered]
ui_1   | chunk    {1} styles.bundle.css, styles.bundle.map, styles.bundle.map (styles) 69.5 kB {3} [initial] [rendered]
ui_1   | chunk    {2} vendor.bundle.js, vendor.bundle.map (vendor) 4.75 MB [initial] [rendered]
ui_1   | chunk    {3} inline.bundle.js, inline.bundle.map (inline) 0 bytes [entry] [rendered]
ui_1   |
ui_1   | ERROR in /app/src/app/chart-details/chart-details-readme/chart-details-readme.component.ts (2,22): Cannot find module 'marked'.
ui_1   |
ui_1   | ERROR in /app/src/app/chart-details/chart-details-info/chart-details-info.component.ts (6,21): Cannot find module 'url-join'.
ui_1   | webpack: Failed to compile.
```

Really not sure why these errors keep coming up...

**Comment by prydonius** [comment]
@kemkake I managed to get rid of the error I was getting above by installing the typings (https://www.reddit.com/r/Angular2/comments/59b2mc/import_vs_require_why_one_work_and_the_other/) and changing the `import` line.

```
npm install --save-dev @types/marked
npm install --save-dev @types/url-join
```

```
diff --git a/src/ui/package.json b/src/ui/package.json
index 5e5fe7a..c7e28ac 100644
--- a/src/ui/package.json
+++ b/src/ui/package.json
@@ -46,6 +46,8 @@
   },
   "devDependencies": {
     "@types/jasmine": "^2.2.30",
+    "@types/marked": "0.0.28",
+    "@types/url-join": "^0.8.2",
     "angular-cli": "1.0.0-beta.24",
     "bootstrap-sass": "^3.3.7",
     "codelyzer": "~0.0.26",
@@ -57,6 +59,7 @@
     "karma-cli": "^1.0.1",
     "karma-jasmine": "^1.0.2",
     "karma-remap-istanbul": "^0.2.1",
+    "mappy-breakpoints": "^0.2.3",
     "node-sass": "^3.9.3",
     "protractor": "4.0.5",
     "resolve-url-loader": "^1.6.0",
@@ -66,7 +69,6 @@
     "tslint": "3.13.0",
     "typescript": "^2.0.0",
     "url-loader": "^0.5.7",
-    "mappy-breakpoints": "^0.2.3",
     "webpack": "2.2.0-rc.3"
   }
 }
diff --git a/src/ui/src/app/chart-details/chart-details-info/chart-details-info.component.ts b/src/ui/src/app/chart-details/chart-details-info/chart-details-info.component.ts
index 115453a..51445c1 100644
--- a/src/ui/src/app/chart-details/chart-details-info/chart-details-info.component.ts
+++ b/src/ui/src/app/chart-details/chart-details-info/chart-details-info.component.ts
@@ -3,7 +3,7 @@ import { ChartsService } from '../../shared/services/charts.service';
 import { Chart } from '../../shared/models/chart';
 import { Maintainer } from '../../shared/models/maintainer';
 import { ChartVersion } from '../../shared/models/chart-version';
-import urljoin from 'url-join';
+import * as urljoin from 'url-join';

 @Component({
   selector: 'app-chart-details-info',
diff --git a/src/ui/src/app/chart-details/chart-details-readme/chart-details-readme.component.ts b/src/ui/src/app/chart-details/chart-details-readme/chart-details-readme.component.ts
index 840b0c8..4f3022f 100644
--- a/src/ui/src/app/chart-details/chart-details-readme/chart-details-readme.component.ts
+++ b/src/ui/src/app/chart-details/chart-details-readme/chart-details-readme.component.ts
@@ -1,5 +1,5 @@
 import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
-import markdown from 'marked';
+import * as markdown from 'marked';
 import { Chart } from '../../shared/models/chart';
 import { ChartsService } from '../../shared/services/charts.service';
 import { ChartVersion } from '../../shared/models/chart-version';
```

**Comment by prydonius** [comment]
If the above works in your environment too, can we update to it and then land? We can revisit these issues when we move start from the new angular project with yarn.
